File: 3251-MaximumAreaOfLongestDiagonalRectangle/3251-MaximumAreaOfLongestDiagonalRectangle.cpp
// Last updated: 8/14/2026, 2:32:55 PM
class Solution {
public:
    int areaOfMaxDiagonal(vector<vector<int>>& dimensions) {
        double maxDiag = 0;
        int maxArea = 0;
        
        for (auto &d : dimensions) {
            int l = d[0], w = d[1];
            double diag = sqrt((double)l * l + (double)w * w);
            int area = l * w;
            
            if (diag > maxDiag || (fabs(diag - maxDiag) < 1e-9 && area > maxArea)) {
                maxDiag = diag;
                maxArea = area;
            }
        }
        return maxArea;
    }
};

File: 3251-MaximumAreaOfLongestDiagonalRectangle/3251-MaximumAreaOfLongestDiagonalRectangle.java
// Last updated: 8/14/2026, 2:32:46 PM
class Solution {
    public int areaOfMaxDiagonal(int[][] dimensions) {
        int maxArea = 0;
        int maxDiagonal = 0;

        for (int[] rect : dimensions) {
            int length = rect[0];
            int width = rect[1];

            int diagonal = length * length + width * width;
            int area = length * width;

            if (diagonal > maxDiagonal || (diagonal == maxDiagonal && area > maxArea)) {
                maxDiagonal = diagonal;
                maxArea = area;
            }
        }

        return maxArea;
    }
}